packages/react-data-grid/src/masks/InteractionMasks.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {cellEventListener} from '../cellEvents/CellEventListener';
+import SelectionMask from './SelectionMask';
+import CellEventTypes from '../cellEvents/CellEventTypes';
+
+class InteractionMasks extends React.Component {
+
+  static propTypes = {
+    visibleStart: PropTypes.number,
+    visibleEnd: PropTypes.number,
+    columns: PropTypes.array,
+    width: PropTypes.number,
+    height: PropTypes.number,
+    onHitBottomBoundary: PropTypes.func,
+    onHitTopBoundary: PropTypes.func,
+    selectedPosition: PropTypes.object,
+    cellEvents: PropTypes.func,
+    rowHeight: PropTypes.number
+  }
+  state = {
+    selectedPosition: {
+      idx: -1,
+      rowIdx: -1
+    }
+  };
+
+  componentDidMount() {
+    const {cellEvents} = this.props;
+    cellEvents.subscribe(CellEventTypes.onClick, (cell) => {
+      this.selectCell(cell);
+      this.node.focus();
+    });
+  }
+
+  onKeyDown = (e) => {
```

Comment:
A decision to move events to a single source of true means that we need to have a single pattern to attach events, all events need to be registered by subscription